packages/flutter_tools/gradle/src/main/kotlin/dependency_version_checker.gradle.kts

```diff
@@ -0,0 +1,270 @@
+        }
+
+        fun checkGradleVersion(version : Version, project : Project) {
+            if (version < errorGradleVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "Gradle",
+                        version.toString(),
+                        errorGradleVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnGradleVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Gradle",
+                        version.toString(),
+                        warnGradleVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkJavaVersion(version : JavaVersion, project : Project) {
+            if (version < errorJavaVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "Java",
+                        version.toString(),
+                        errorJavaVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnJavaVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Java",
+                        version.toString(),
+                        warnJavaVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkAGPVersion(version : Version, project : Project) {
+            if (version < errorAGPVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "AGP",
+                        version.toString(),
+                        errorAGPVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnAGPVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "AGP",
+                        version.toString(),
+                        warnAGPVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkKGPVersion(version : Version, project : Project) {
+            if (version < errorKGPVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "Kotlin",
+                        version.toString(),
+                        errorKGPVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnKGPVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Kotlin",
+                        version.toString(),
+                        warnKGPVersion.toString()
+                    )
+                )
+            }
+        }
+    }
+}
+
+
+// Helper class to parse the versions that are provided as plain strings (Gradle, Kotlin) and
```

Comment:
Document how rc candidates are treated in ktdoc. 
Also document how shorter versions like 8.2 are treated. 

For example is 8.2 == 8.2.0?

Reply:
Added an explanation